src/lib/peterportal.types.ts

```diff
@@ -127,3 +127,9 @@ export interface SectionGQL {
     number: string;
     type: string;
 }
+
+export interface PrerequisiteJSON {
```

Comment:
nit: I don't like this variable name. Maybe call this PrerequisiteTreeNode and call the other class PrerequisiteTreeValue? Or call this one PrereuquisiteTreeRecord or something. Usually JSON in the name of something implies data being passed as strings in JSON format somewhere but we aren't doing encoding/decoding at all here.